### ISSUE TITLE
Handlers Are Not Enforcing PLM Acks

### ISSUE THREAD

**Issue by krkeegan** [TD22057/insteon-mqtt#260]
We parse the ACKs just fine, but the handlers don't ensure that a PLM ACK arrives before parsing the response from the device.  In theory this shouldn't ever be an issue if our timings are right.  But there could be an echo of a prior message that could arrive after we queue the message in the PLM?  I don't know how the PLM would respond to this and it isn't easily testable.  I am inclined to think that it can't happen, but sillier things have been over looked.

I think adding a simple check the handlers before processing a device response is an easy addition.

**Comment by krkeegan** [comment]
I am not going to worry about this right now since it doesn't seem to be causing an issue.

**Comment by krkeegan** [comment]
Ha, as soon as I dismiss it, I finally spotted an error.  This error in part arose because my automation is spamming a device on command way too frequently, I need to discard commands in a window.  But still this shouldn't cause IM to crash.

In summary, this is what happened:
1. A series of devices all triggered an automation 2 times in a little less than a second.
2. This caused 2 `on` commands to get queued for a device.
3. The first `on` command was sent to the PLM.
4. The PLM acks this command.
3. Then a motion sensor also fired this caused:
  a. the second `on` command to be queued.  
  b. a battery request to be queued.
  c. increased the write delay time.
4. The first `on` device ACK then received with max hops 2 and 1 hop left at 126 milliseconds
5. The second `on` command is sent to the Serial object, but due to the write delay, is not yet written.
6. Then a second `on` device ACK is then received with max hops 2 and 0 hops left at 222 milliseconds.
7. This second `on` ACK from the first command is then sent to handler for the second `on` command.
  a. It processes it and marks the queue as clear.
8. The `get battery voltage` message is then sent to the Serial object
9. The write time delay then elapses.
10. BOTH the `on` and `get battery voltage` messages are sent at basically the same time.
11. The second message causes an error in the assertion in `Protocol._msg_written()` because the link is not in `PENDING_WRITE` status but is actually in `WAIT_FOR_REPLY`

tl;dr; Step 7 is where things break.  A handler interprets an ACK is valid, even though the outgoing message was never sent, let alone Ack'd by the PLM.

Lots of things could fix this.  As noted, I should add anti-spam to my automation.  Inside IM I think we need to:
1. Check that the message was written via the call to `sending_message` before processing a PLM ACK
2. Check that the PLM Ack was received prior to processing a device ACK

Fix 1 is essential, there is no possibility that a PLM ACK can arrive before the message is written.

I believe Fix 2 is also safe, over ten years, I have no recollection of ever seeing a PLM ACK getting lost.  Even in the extremely rare possibility that it happens, the command would simply time-out and retry.

In the weeds note: We attempt to catch duplicate messages, so the message in Step 6 above should have been caught as a duplicate, but it arrived too late 96 milliseconds instead of within the 87 milliseconds we currently use.  I suspect that 99% of the time this would have been caught, the insteon spec says this time should be 50 milliseconds.  I can only assume that all of the traffic somehow caused it to be delayed?
